server/game/cards/14-FotS/ACaskOfAle.js

```diff
@@ -5,56 +5,71 @@ class ACaskOfAle extends DrawCard {
     setupCardAbilities() {
         this.reaction({
             when: {
-                afterChallenge: event => event.challenge.isMatch({ winner: this.controller, challengeType: 'power' })
+                afterChallenge: event => (event.challenge.isMatch({ winner: this.controller, challengeType: 'power' }) 
+                    && this.game.allCards.some(card => (card.location === 'active plot', 'faction', 'play area' && card.getPower()>0)))
```

Comment:
The location check is not quite valid syntax. I think you want something more like `['active plot', 'faction', 'play area'].includes(card.location)` instead (similar to what you did below)